sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueClientBuilder.java

```diff
@@ -193,7 +193,16 @@ public QueueClientBuilder() {
      * and {@link #retryOptions(RequestRetryOptions)} have been set.
      */
     public QueueClient buildClient() {
-        return new QueueClient(buildAsyncClient());
+        StorageImplUtils.assertNotNull("queueName", queueName);
+        if (processMessageDecodingErrorAsyncHandler != null && processMessageDecodingErrorHandler != null) {
```

Comment:
Should we check if the sync error handler is set when `buildClient` is called and similarly check that the async error handler is set when `buildAsyncClient` is called? It may be a behavior change but using a sync error handler in async client can cause some issues with blocking threads in a reactive stream.